crates/storage/src/connection/block.rs

```diff
@@ -388,6 +388,57 @@ pub(super) fn block_is_l1_accepted(tx: &Transaction<'_>, block: BlockId) -> anyh
     Ok(block_number <= l1_l2)
 }
 
+pub(super) fn first_block_without_transactions(
+    tx: &Transaction<'_>,
+) -> anyhow::Result<Option<BlockNumber>> {
+    let mut stmt = tx
+        .inner()
+        .prepare(
+            "
+            SELECT number
+            FROM block_headers
+            LEFT JOIN starknet_transactions ON starknet_transactions.block_hash = block_headers.hash
```

Comment:
Not related to this PR at all; but we really should move transactions to using block number instead of hash.